ui-tests/cypress/integration/components/ux_fragments.spec.js

```diff
@@ -1,141 +1,783 @@
+
+            page.getContent().getPagination().get()
+                .should('exist').and('be.visible');
+            page.getContent().getPagination().getInput()
+                .should('have.value', 1);
+
+            page.getContent().getSearchForm()
+                .should('exist').and('be.visible');
+            page.getContent().getAddButton()
+                .should('exist').and('be.visible');
+          });
     });
-
-    it('Add a new fragment', () => {
-      currentPage = openFragmentsPage();
-
-      currentPage = currentPage.getContent().openAddFragmentPage();
+    it([Tag.SMOKE, 'ENG-3522'], 'AddPage is displayed', () => {
+      openFragmentsPage()
+          .then(page =>
+              page.getContent().openAddFragmentPage());
       cy.validateUrlPathname('/fragment/add');
-      currentPage.getContent().typeCode(fragment.code);
-      currentPage.getContent().typeGuiCode(fragment.guiCode);
-      currentPage = currentPage.getContent().save();
-      cy.wrap(fragment).as('fragmentToBeDeleted');
 
-      cy.validateToast(currentPage);
-    });
+      cy.get('@currentPage')
+        .then(page => {
+          page.getContent().getCodeInput().should('exist').and('be.visible');
+          page.getContent().getGuiCodeInput().should('exist').and('be.visible');
 
-    it('Add a new fragment using an existing code - not allowed', () => {
-      cy.fragmentsController()
-        .then(controller => controller.addFragment(fragment))
-        .then(() => cy.wrap(fragment).as('fragmentToBeDeleted'));
+          page.getContent().getGuiCodeSelector().should('exist').and('be.visible').and('have.text', 'Gui code');
+          page.getContent().getDefaultGuiCodeSelector().should('exist').and('be.visible').and('have.text', 'Default gui code');
 
-      currentPage = openFragmentsPage();
-
-      currentPage = currentPage.getContent().openAddFragmentPage();
-      cy.validateUrlPathname('/fragment/add');
-      currentPage.getContent().typeCode(fragment.code);
-      currentPage.getContent().typeGuiCode(fragment.guiCode);
-      currentPage.getContent().save();
-
-      cy.validateToast(currentPage, fragment.code, false);
+          page.getContent().getCancelBtn().should('exist').and('be.visible');
+          page.getContent().getSaveBtn().should('exist').and('be.visible');
+        });
     });
-
-    it('Add a new fragment with an invalid code - not allowed', () => {
-      currentPage = openFragmentsPage();
-
-      currentPage = currentPage.getContent().openAddFragmentPage();
-      cy.validateUrlPathname('/fragment/add');
-      currentPage.getContent().typeCode(`${fragment.code}!@#$%^&*`);
-      currentPage.getContent().typeGuiCode(fragment.guiCode);
-      currentPage.getContent().clickSaveBtn();
-
-      currentPage.getContent().getSaveOption().closest('li').invoke('attr', 'class').should('contain', 'disabled');
+    it([Tag.SMOKE, 'ENG-3522'], 'Dropdown is displayed', () => {
+      addTestFragment();
```

Comment:
you can concatenate the test to `addFragment` and propagate the created value instead of storing it in a `const`